OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -109,19 +108,14 @@ List<IFinalizedRenderable> GenerateRenderables()
 			if (World.OrderGenerator != null)
 				worldRenderables = worldRenderables.Concat(World.OrderGenerator.Render(this, World));
 
+			worldRenderables = worldRenderables.Concat(World.Effects.SelectMany(e => e.Render(this)));
 			worldRenderables = worldRenderables.OrderBy(RenderableScreenZPositionComparisonKey);
 
-			// Effects are drawn on top of all actors
-			// HACK: Effects aren't interleaved with actors.
-			var effectRenderables = World.Effects
-				.SelectMany(e => e.Render(this));
-
 			if (World.OrderGenerator != null)
-				effectRenderables = effectRenderables.Concat(World.OrderGenerator.RenderAfterWorld(this, World));
+				worldRenderables = worldRenderables.Concat(World.OrderGenerator.RenderAfterWorld(this, World));
```

Comment:
This is best left to its own PR.  It's already going to take a lot of work to verify the effect ordering, and order generators do weird things with sprites.